src/drivers/imx/interrupt-irqstr.c

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2021 NXP
+//
+// Author: Peng Zhang <peng.zhang_8@nxp.com>
+
+#include <sof/common.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/drivers/irqstr.h>
+#include <sof/lib/cpu.h>
+#include <sof/lib/io.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/spinlock.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+void platform_interrupt_init(void)
+{
+	int i;
+	extern const struct irq_cascade_tmpl dsp_irq[DSP_IRQ_MAX];
+
+	/* Turn off the hardware so we don't have stray interrupts while
+	 * initializing
+	 */
+	irqstr_disable_hw();
+	/* Mask every external IRQ first */
+	for (i = 0; i < IRQSTR_IRQS_REGISTERS_NUM; i++)
+		irqstr_write(IRQSTR_CH_MASK(i), 0);
+	/* Turn on the IRQ_STEER hardware */
+	irqstr_enable_hw();
+
+	for (i = 0; i < ARRAY_SIZE(dsp_irq); i++)
+		interrupt_cascade_register(dsp_irq + i);
```

Comment:
Might want to refactor the body of this function into a separate irqstr_init(), and platform_interrupt_init just calls irqstr_init().